ouroboros-consensus/src/Ouroboros/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -306,7 +310,11 @@ chainSelectionForBlock cdb@CDB{..} blockCache hdr = do
     if
       -- The chain might have grown since we added the block such that the
       -- block is older than @k@.
-      | At (blockNo hdr) <= immBlockNo && blockNo hdr /= 0 ->
+      | At (blockNo hdr) <= immBlockNo &&
+        -- Exception: the block we're adding is the EBB with the same block
+        -- number as the immutable tip. This can only happen when the
+        -- VolatileDB was corrupted.
+        not (At (blockNo hdr) == immBlockNo && cdbIsEBB hdr == IsEBB) ->
```

Comment:
Should we introduce an `olderThanK` predicate, document it with this comment, and then use that here and above rather than duplicate this logic?